src/libraries/System.Memory/tests/ArrayBufferWriter/ArrayBufferWriterTests.Byte.cs

```diff
@@ -106,12 +106,21 @@ public void GetMemory_ExceedMaximumBufferSize()
             output.Advance(initialCapacity);
 
             // Validate we can't double the buffer size, but can grow
-            Memory<byte> memory = output.GetMemory(1);
-
-            // The buffer should grow more than the 1 byte requested otherwise performance will not be usable
-            // between 1GB and 2GB. The current implementation maxes out the buffer size to MaxArrayLength.
-            Assert.Equal(MaxArrayLength - initialCapacity, memory.Length);
-
+            Memory<byte> memory;
+            try
+            {
```

Comment:
If allocating ~2 GB is problematic, could ~1 GB also be an issue? In that case, it might be better to put the try/catch around the whole test, including the initialization of ABW and the first call to Advance.

Alternatively, maybe we should just opt to skip the test for the specific constrained devices where 1-2GB allocation is problematic (with some kind of conditional skip attribute). How much RAM is available on the Windows.Nano test machines?

Reply:
Perhaps 1GB could cause issues as well; I see other JSON inner loop tests that allocate ~300K but not up to 1GB.